course/src/isometry.cc

```diff
@@ -312,5 +312,79 @@ namespace ekumen {
     return Vector3(row1_[index], row2_[index], row3_[index]);
   }
 
+  // Isometry
+
+  bool Isometry::operator == (const Isometry & obj) const
+  {
+    return ((rotation_ == obj.rotation()) && (point_ == obj.point()));
+  }
+
+  Vector3 Isometry::operator *= (const Vector3 & obj) const
+  {
+    return Vector3(rotation_ * obj + point_);
+  }
+
+  Isometry Isometry::operator *= (const Isometry & obj) const
+  {
+    Matrix3 new_rotation = rotation_ * obj.rotation();
+    Vector3 new_point = rotation_ * obj.point() + point_;
+    return Isometry(new_point, new_rotation);
+  }
+
+  Vector3 Isometry::transform (const std::initializer_list<double> & obj) const
+  {
+    Vector3 temp(obj);
+    return Vector3(rotation_ * temp + point_);
+  }
+
+  Isometry Isometry::inverse () const
+  {
+    double det = rotation_.det();
+    if (det == 0)
+    {
+      throw std::runtime_error("Matrix is not invertable");
+    }
+    Matrix3 new_rotation(
+        (rotation_[1][1] * rotation_[2][2] - rotation_[1][2] * rotation_[2][1]) / det,
```

Comment:
Please implement the inverse of a 3x3 matrix in Matrix3, and use that. It's easier to test.